### PR TITLE
Update docs to mention new additions in Compose 1.9.0

### PR DESCRIPTION
**Pull request by shin-**
### Project version

Compose 1.9.0
### Related issue

N/A
### Related issue or PR in another project

https://github.com/docker/compose/pull/4048
### Please take a look

@aanand @dnephin

### CODE REVIEW

**Comment by dnephin** [comment]
LGTM

File: compose/compose-file.md [hunk 1]
```diff
@@ -409,6 +434,15 @@ options and tags it with the specified tag.
...
 > with `image` is not allowed. Attempting to do so results in an error.
 
+### isolation
+
+> [Added in version 2.1 file format](#version-21).
+
+Specify a container’s isolation technology. On Linux, the only supported value
+is `default`. On Windows, acceptable values are `default`, `process` and
+`hyperv`. Refer to the
+[Docker Engine docs](https://docs.docker.com/engine/reference/commandline/run/#specify-isolation-technology-for-container---isolation) for details.
```
Comment (johndmulhausen) [inline compose/compose-file.md]: Can you change the URL to start at `/engine/` here? That way this link will work in staging, doc archives, etc. :)

File: compose/compose-file.md [hunk 2]
```diff
@@ -888,14 +978,41 @@ A full example:
...
+
+By default, Docker also connects a bridge network to it to provide external
+connectivity. If you want to create an externally isolated overlay network,
+you can set this option to `true`.
+
+### labels
+
+> [Added in version 2.1 file format](#version-21).
+
+Add metadata to containers using [Docker labels](https://docs.docker.com/engine/userguide/labels-custom-metadata/). You can use either an array or a dictionary.
```
Comment (johndmulhausen) [inline compose/compose-file.md]: Ditto on URL starting at /engine/

File: compose/compose-file.md [hunk 3]
```diff
@@ -391,6 +391,31 @@ An entry with the ip address and hostname will be created in `/etc/hosts` inside
...
+
+> [Version 2 file format](compose-file.md#version-2) and up.
+
+Specify additional groups (by name or number) which the user inside the
+container will be a member of. Groups must exist in both the container and the
+host system to be added. An example of where this is useful is when multiple
+containers (running as different users) need to all read or write the same
+file on the host system. That file can be owned by a group shared by all the
+containers, and specified in `group_add`. See the
+[Docker documentation](https://docs.docker.com/engine/reference/run/#additional-groups)
```
Comment (johndmulhausen) [inline compose/compose-file.md]: URL should start at /engine/

File: compose/compose-file.md [hunk 4]
```diff
@@ -409,6 +434,15 @@ options and tags it with the specified tag.
 > **Note**: In the [version 1 file format](compose-file.md#version-1), using `build` together
 > with `image` is not allowed. Attempting to do so results in an error.
 
+### isolation
+
+> [Added in version 2.1 file format](#version-21).
```
Comment (johndmulhausen) [inline compose/compose-file.md]: Are you having luck specifying just the `#tag` without the filename in it? I think in order for GitHub's markdown previewer AND our rendered site work, it is necessary for anchor links to include the file name, due to our use of relative links.  Sorry. :/ 

See: The other examples with URLs like `compose-file.md#tag`

Try staging the docs (just run `docker-compose up` in the root of your clone of this repo) and see if these links work on your localhost server. If they do, great; if not, this PR has a few tweaks in store.

**Comment by shin-** [comment]
@johndmulhausen Thanks for the review! I updated the PR to address your comments, and I added a paragraph for the improvements we made on [variable substitution](https://github.com/docker/docker.github.io/pull/285/files#diff-7f92e63d2e7e84a65a6497458b96739cR1285).

**Comment by shin-** [comment]
@johndmulhausen @londoncalling - Rebased!

**Comment by londoncalling** [comment]
LGTM

**Pull request merged**